### PR TITLE
Add new project for upstream tinkerbell chart and tinkerbell crds chart

### PR DESCRIPTION
**Pull request by ahreehong**
*Issue #, if available:*

*Description of changes:*
Adding new project to build the upstream tinkerbell chart, as well as build a separate tinkerbell crds chart.g

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by ahreehong** [comment]
holding until CDK changes are merged

File: projects/tinkerbell/charts/docker/linux/nginx/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,30 @@
+ARG BASE_IMAGE # https://gallery.ecr.aws/eks-distro-build-tooling/eks-distro-minimal-base-nginx
+ARG BUILDER_IMAGE
+
+FROM $BUILDER_IMAGE as builder
+
+RUN set -x && \
+    install_binary /usr/sbin/awk && \
+    cleanup "deps"
+
+RUN set -x && \
```
Comment (jaxesn) [inline projects/tinkerbell/charts/docker/linux/nginx/Dockerfile:10]: this talk about this, same with the /dev/stdout stuff below

File: projects/tinkerbell/charts/docker/linux/dhcrelay/docker/linux/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+ARG BUILDER_IMAGE
+FROM $BUILDER_IMAGE as builder
+
+RUN set -x && \
+    install_binary /usr/sbin/dhcrelay && \
+    cleanup "deps"
+
+FROM $BASE_IMAGE
+
+ARG RELEASE_BRANCH
```
Comment (jaxesn) [inline projects/tinkerbell/charts/docker/linux/dhcrelay/docker/linux/Dockerfile:11]: why are these here?

File: projects/tinkerbell/charts/Makefile [hunk 3]
```diff
@@ -0,0 +1,31 @@
+BASE_DIRECTORY:=$(abspath ../../../)
```
Comment (jaxesn) [inline projects/tinkerbell/charts/Makefile:1]: why is this a separate project? in other projects with helm charts we include it in the main project build

File: projects/tinkerbell/charts/Makefile [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+GIT_TAG=$(shell cat GIT_TAG)
+GOLANG_VERSION=$(shell cat GOLANG_VERSION)
+REPO=charts
+REPO_OWNER=tinkerbell
+
+IMAGE_NAMES=nginx
+DOCKERFILE_FOLDER=./docker/linux/$(IMAGE_NAME)
```
Comment (jaxesn) [inline projects/tinkerbell/charts/Makefile:8]: is this needed? is this not the default?

File: projects/tinkerbell/charts/Makefile [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+
+NGINX_IMAGE_COMPONENT=tinkerbell/charts/nginx
```
Comment (jaxesn) [inline projects/tinkerbell/charts/Makefile:10]: i find this a bit weird to be "under" the charts

File: Common.mk [hunk 6]
```diff
@@ -860,10 +861,10 @@ $(call FULL_CHART_TARGETS,require) : %/helm/require: %/helm/copy | $$(ENABLE_LOG
 	@$(BUILD_LIB)/helm_require.sh $(HELM_SOURCE_IMAGE_REPO) $(HELM_DESTINATION_REPOSITORY) $(OUTPUT_DIR) $(IMAGE_TAG) $(HELM_TAG) $(PROJECT_ROOT) $(LATEST) $(HELM_USE_UPSTREAM_IMAGE) "$(PACKAGE_DEPENDENCIES)" "$(FORCE_JSON_SCHEMA_FILE)" "$(HELM_IMAGE_LIST)" "$(HELM_IMAGE_TAG_LIST)"
 
 $(call FULL_CHART_TARGETS,replace) : %/helm/replace: %/helm/require | $$(ENABLE_LOGGING)
-	@$(BUILD_LIB)/helm_replace.sh $(HELM_DESTINATION_REPOSITORY) $(OUTPUT_DIR)
+	@$(BUILD_LIB)/helm_replace.sh $(CHART_NAME) $(OUTPUT_DIR)
```
Comment (jaxesn) [inline Common.mk:864]: i believe this is the only spot, but make sure these scripts are only called here. this helm stuff used to be all over the place. this seems like a fair change, but what made you make it?

File: projects/tinkerbell/charts/docker/linux/dhcrelay/docker/linux/Dockerfile [hunk 7]
```diff
@@ -0,0 +1,15 @@
+ARG BASE_IMAGE # https://gallery.ecr.aws/eks-distro-build-tooling/eks-distro-minimal-base-glibc
```
Comment (jaxesn) [inline projects/tinkerbell/charts/docker/linux/dhcrelay/docker/linux/Dockerfile:1]: the file path/name is odd.

File: projects/tinkerbell/charts/docker/linux/relay-init/Dockerfile [hunk 8]
```diff
@@ -0,0 +1,4 @@
+ARG BASE_IMAGE # https://gallery.ecr.aws/eks-distro-build-tooling/eks-distro-minimal-base-nsenter
```
Comment (jaxesn) [inline projects/tinkerbell/charts/docker/linux/relay-init/Dockerfile:1]: what is relay-init used for?

File: projects/tinkerbell/charts/helm/sedfile.template [hunk 9]
```diff
@@ -0,0 +1 @@
+s,0.4.4,${IMAGE_TAG},g
```
Comment (jaxesn) [inline projects/tinkerbell/charts/helm/sedfile.template:1]: i dont think this is your fault, but this stuff is super scary, seems easy to miss. make sure the updating instructions calls this out clearly.

File: projects/tinkerbell/tinkerbell-crds/chart/templates/rufio/bmc.tinkerbell.org_jobs.yaml [hunk 10]
```diff
@@ -4,6 +4,7 @@ kind: CustomResourceDefinition
 metadata:
```
Comment (jaxesn) [inline projects/tinkerbell/tinkerbell-crds/chart/templates/rufio/bmc.tinkerbell.org_jobs.yaml:4]: we do we have a seperate chart for crds? does the upstream chart not include these?

**Review by jaxesn — CHANGES_REQUESTED** [review]
I have a number of questions, Id like to go through this with you.

File: projects/tinkerbell/charts/docker/linux/dhcrelay/docker/linux/Dockerfile (hunk 7 above)
Comment (ahreehong) [reply projects/tinkerbell/charts/docker/linux/dhcrelay/docker/linux/Dockerfile:1]: this file shouldn't be included in the pr, I'll remove it

File: projects/tinkerbell/charts/docker/linux/relay-init/Dockerfile (hunk 8 above)
Comment (ahreehong) [reply projects/tinkerbell/charts/docker/linux/relay-init/Dockerfile:1]: Same with this one, i'll remove from the pr. Relay init will be used for https://github.com/tinkerbell/charts/blob/43a6c2ce1056f1d409ddef8ee7d45c33138e763c/tinkerbell/stack/templates/nginx.yaml#L157

File: Common.mk [hunk 11]
```diff
@@ -212,6 +212,7 @@ HELM_USE_UPSTREAM_IMAGE?=false
 # HELM_DIRECTORY must be a relative path from project root to the directory that contains a chart
 HELM_DIRECTORY?=.
 HELM_DESTINATION_REPOSITORY?=$(HELM_CHART_NAME)
+CHART_NAME?=$(shell basename $(HELM_DESTINATION_REPOSITORY))
```
Comment (jaxesn) [inline Common.mk:215]: Lets call this HELM_CHART_FOLDER?

**Review by jaxesn — APPROVED** [review]
Looks good!

**Comment by ahreehong** [comment]
/override tinkerbell-crds-presubmit
This is failing looking for the charts.yaml on the main branch to verify against, but this is a new chart so the file does not exist.
